web_app/tabs/i3_mno.py

```diff
@@ -175,252 +191,385 @@ def layout() -> html.Div:
+        Output("tab-1-make-snapshot-button", "disabled"),
+        Output("tab-1-how-to-edit-message", "children"),
+        Output("tab-1-data-table", "row_deletable"),
+    ],
     [Input("tab-1-input-name", "value"), Input("tab-1-input-email", "value")],
 )
-def update_name_email_icon(name: str, email: str) -> str:
+def sign_in(name: str, email: str) -> Tuple[str, bool, bool, bool, bool, str, bool]:
     """Enter name & email callback."""
     # TODO -- check auth
+
     if name and email:
-        return "✔"
-    return "✖"
+        return (
+            "✔",
+            True,  # data-table editable
+            False,  # new-data-button-top NOT disabled
+            False,  # new-data-button-bottom NOT disabled
+            False,  # make-snapshot-button NOT disabled
+            "click a cell to edit",
+            True,  # row is deletable
+        )
+    return (
+        "✖",
+        False,  # data-table NOT editable
+        True,  # new-data-button-top disabled
+        True,  # new-data-button-bottom disabled
+        True,  # make-snapshot-button disabled
+        "sign in to edit",
+        False,  # row NOT deletable
+    )
 
 
 @app.callback(  # type: ignore[misc]
-    Output("tab-1-dropdown-per-row", "data"),
-    [
-        Input("tab-1-filter-dropdown-inst", "value"),
-        Input("tab-1-filter-dropdown-labor", "value"),
-        Input("tab-1-editing-rows-button", "n_clicks"),
-    ],
     [
-        State("tab-1-dropdown-per-row", "data"),
-        State("tab-1-dropdown-per-row", "columns"),
+        Output("tab-1-show-all-rows-button", "children"),
+        Output("tab-1-show-all-rows-button", "color"),
+        Output("tab-1-show-all-rows-button", "outline"),
+        Output("tab-1-data-table", "page_size"),
+        Output("tab-1-data-table", "page_action"),
     ],
+    [Input("tab-1-show-all-rows-button", "n_clicks")],
 )
-def display_table(
-    institution: List[Dict[str, str]],
-    labor: List[Dict[str, str]],
-    n_clicks: int,
-    rows: List[Dict[str, str]],
-    columns: List[Dict[str, str]],
-) -> List[Dict[str, str]]:
-    """Filter dropdown menu & new rows callbacks."""
-    ctx = dash.callback_context
-
-    if not ctx.triggered:
-        button_id = "None"
-    else:
-        button_id = ctx.triggered[0]["prop_id"].split(".")[0]
-
-    if n_clicks == 0 or button_id in ("filter-dropdown-inst", "filter-dropdown-labor"):
-        # print(button_id)
-        if not institution:
-            dff = DF[DF["Labor Cat."] == labor]
-        elif not labor:
-            dff = DF[DF["Institution"] == institution]
-        else:
-            dff = DF[DF["Institution"] == institution]
-            dff = dff[dff["Labor Cat."] == labor]
-        return dff.to_dict("records")
+def toggle_pagination(n_clicks: int) -> Tuple[str, str, bool, int, str]:
+    """Toggle whether the table is paginated."""
+    if n_clicks % 2 == 0:
+        return "Show All Rows", "secondary", True, 15, "native"
```

Comment:
This is where the default number of rows comes from?

I think 15 seems small, but that's just me.

Reply:
That's right. I'll make it configurable